README.md

```diff
@@ -56,13 +56,13 @@ For further customization, see the configuration for `rebar3_ex_doc`:
 ```erlang
 {ex_doc, [
     {extras, [
-          {'CHANGELOG.md', #{title => <<"Changelog">>}},
-          {'README.md', #{title => <<"Overview">>}},
-          {'LICENSE.md', #{title => <<"License">>}}
+          {"CHANGELOG.md", #{title => "Changelog"}},
+          {"README.md", #{title => "Overview"}},
+          {"LICENSE.md", #{title => "License"}}
     ]},
-    {main, <<"readme">>},
-    {source_url, <<"https://github.com/starbelly/rebar3_ex_doc">>},
-    {assets, <<"assets">>},
+    {main, "README.md"},
+    {source_url_pattern, "https://github.com/starbelly/rebar3_ex_doc"},
```

Comment:
Why rename this to `source_url_pattern` ?

Reply:
I'm confused. I thought `ex_doc` (from the plugin) would correspond to `docs` in the "mix" file. Is that not it?

Reply:
If you mean the mix file... the way you define this in a mix.exs is like so: 

```
def project do
  [
    app: :my_app,
    version: "0.1.0-dev",
    deps: deps(),

    # Docs
    name: "My App",
    source_url: "https://github.com/USER/PROJECT",
    homepage_url: "http://YOUR_PROJECT_HOMEPAGE",
    docs: [
      main: "MyApp", # The main page in the docs
      logo: "path/to/logo.png",
      extras: ["README.md"]
    ]
  ]
```

You can also refer to ex_doc (escript) itself : 

```
sage:
  ex_doc PROJECT VERSION BEAMS [OPTIONS]

Examples:
  ex_doc "Ecto" "0.8.0" "_build/dev/lib/ecto/ebin" -u "https://github.com/elixir-ecto/ecto"
  ex_doc "Project" "1.0.0" "_build/dev/lib/project/ebin" -c "docs.exs"

Options:
  PROJECT             Project name
  VERSION             Version number
  BEAMS               Path to compiled beam files
  -n, --canonical     Indicate the preferred URL with rel="canonical" link element
  -c, --config        Give configuration through a file instead of a command line.
                      See "Custom config" section below for more information.
  -f, --formatter     Docs formatter to use (html or epub), default: "html"
  -p, --homepage-url  URL to link to for the site name
      --paths         Prepends the given path to Erlang code path. The path might contain a glob
                      pattern but in that case, remember to quote it: --paths "_build/dev/lib/*/ebin".
                      This option can be given multiple times
      --language      Identify the primary language of the documents, its value must be
                      a valid [BCP 47](https://tools.ietf.org/html/bcp47) language tag, default: "en"
  -l, --logo          Path to the image logo of the project (only PNG or JPEG accepted)
                      The image size will be 64x64 and copied to the assets directory
  -m, --main          The entry-point page in docs, default: "api-reference"
      --package       Hex package name
      --source-ref    Branch/commit/tag used for source link inference, default: "master"
  -u, --source-url    URL to the source code
  -o, --output        Path to output docs, default: "doc"
  -v, --version       Print ExDoc version

## Custom config

A custom config can be given with the `--config` option.

The file must either have ".exs" or ".config" extension.

The file with the ".exs" extension must be an Elixir script that returns a keyword list with
the same options declares in `Mix.Tasks.Docs`. Here is an example:

    [
      extras: Path.wildcard("lib/elixir/pages/*.md"),
      groups_for_functions: [
        Guards: & &1[:guard] == true
      ],
      skip_undefined_reference_warnings_on: ["compatibility-and-deprecations"],
      groups_for_modules: [
        ...
      ]
    ]

The file with the ".config" extension must contain Erlang terms separated by ".".
See `:file.consult/1` for more information. Here is an example:

    {extras, [<<"README.md">>, <<"CHANGELOG.md">>]}.
    {main, <<"readme">>}.
    {proglang, erlang}.

## Source linking

ExDoc by default provides links to the source code implementation as
long as `--source-url` or `--source-url-pattern` is provided. If you
provide `--source-url`, ExDoc will inflect the url pattern automatically
for GitHub, GitLab, and Bitbucket URLs. For example:

    --source-url "https://github.com/elixir-ecto/ecto"

Will be inflected as:

    https://github.com/elixir-ecto/ecto/blob/master/%{path}#L%{line}

To specify a particular branch or commit, use the `--source-ref` option:

    --source-url "https://github.com/elixir-ecto/ecto" --source-ref "v1.0"

will result in the following URL pattern:

    https://github.com/elixir-ecto/ecto/blob/v1.0/%{path}#L%{line}
    ```

Reply:
Sure, which is why I believe the `source_url` and `source_url_pattern` are not at the same level. (maybe there's some magic between the plugin and the "mix file" (?))